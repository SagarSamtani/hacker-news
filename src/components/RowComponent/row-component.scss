@import "../../css/global/vairables.scss";
// @import "../../css/global/mixins.scss";

.row-container {
  display: flex;
  align-items: center;
  font-size: 0.6875rem;

  .number-of-comments {
    width: 90px;
    justify-content: center;
  }
  .title {
    font-size: 0.8125rem;
  }
  .upvote-arrow-img {
    height: 8px;
    width: 8px;
    margin: 5px 5px 0;
    cursor: pointer;
  }

  a:link {
    color: $black-color;
    text-decoration: none;
  }
  a:visited {
    color: $grey-text-color;
    text-decoration: none;
  }
  .link-domain {
    cursor: pointer;
    color: #828890;
  }
  .posted-when {
    color: #828890;
  }
  .username {
    color: #b8b9bb;
  }
  .author {
    color: $black-color;
  }
  .row-elements {
    display: flex;
    flex-flow: row wrap;
    padding: 3px;
  }
  .upvotes-count {
    width: 25px;
    text-align: right;
  }
  .hide {
    cursor: pointer;
  }
  .ml-5 {
    margin-left: 5px;
  }
}
.row-container:nth-child(even) {
  background-color: #e6e6df;
}
