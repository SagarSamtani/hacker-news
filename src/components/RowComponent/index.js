import React from "react";
import ImageComponent from "../ImageComponent";
import propTypes from "prop-types";
import moment from "moment";
import "./row-component.scss";

const RowComponent = ({
  rowValue,
  upVoteNews,
  handleDomainClick,
  removeNewsItem
}) => {
  const { num_comments, points, title, url, author, created_at } = rowValue;
  let modifiedUrl =
    url &&
    rowValue.url
      .replace("http://", "")
      .replace("https://", "")
      .replace("www.", "")
      .split(/[/?#]/)[0];
  return (
    <div className="row-container">
      <div className="row-elements number-of-comments">
        <span>{num_comments || "-"}</span>
      </div>
      <div className="row-elements polls-upvotes-title-container">
        <span className="upvotes-count"> {points || 0}</span>
        <ImageComponent
          imageClassName="upvote-arrow-img"
          imageName="upvote-arrow-img.gif"
          imageAlt="upvote-arrow-img"
          handleClick={() => {
            upVoteNews(rowValue);
          }}
        />
        <span className="title">
          <a href={url}>{title}</a>
        </span>
      </div>
      <div className="row-elements username-doamin-hide-container">
        <span
          className="link-domain"
          onClick={() => {
            handleDomainClick(modifiedUrl);
          }}
        >
          {modifiedUrl ? `(${modifiedUrl}) ` : "---------"}
        </span>
        <span className="ml-5 username">
          {`by `} <span className="author">{author}</span>
        </span>
        <span className="ml-5 posted-when">{moment(created_at).fromNow()} </span>
        <span className="ml-5 hide" onClick={() => removeNewsItem(rowValue)}>
          [ hide ]
        </span>
      </div>
    </div>
  );
};

RowComponent.propTypes = {
  rowValue: propTypes.object,
  upVoteNews: propTypes.func,
  handleDomainClick: propTypes.func,
  removeNewsItem: propTypes.func
};

export default RowComponent;
