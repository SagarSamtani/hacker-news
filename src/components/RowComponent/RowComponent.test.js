import React from "react";
import RowComponent from "./index";
import { expect } from "chai";
import { shallow } from "enzyme";

const mockRowValue = {
  title: "Y Combinator",
  url: "http://ycombinator.com",
  author: "pg",
  points: 57,
  story_text: null,
  comment_text: null,
  _tags: ["story"],
  num_comments: 2,
  objectID: "1",
  _highlightResult: {
    title: {
      value: "Y Combinator",
      matchLevel: "none",
      matchedWords: []
    },
    url: {
      value: "http://ycombinator.com",
      matchLevel: "none",
      matchedWords: []
    },
    author: {
      value: "<em>pg</em>",
      matchLevel: "full",
      matchedWords: ["pg"]
    }
  }
};

describe("<RowComponent />", () => {
  it("renders correctly", () => {
    const rowCompnent = shallow(<RowComponent rowValue={mockRowValue} />);
    expect(rowCompnent).to.have.lengthOf(1);
  });
  it("renders an `.upvotes-count`", () => {
    const wrapper = shallow(<RowComponent rowValue={mockRowValue} />);
    expect(wrapper.find(".upvotes-count")).to.have.lengthOf(1);
  });
});
