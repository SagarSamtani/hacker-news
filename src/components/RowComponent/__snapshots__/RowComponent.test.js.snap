// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyComponent /> renders correctly 1`] = `ShallowWrapper {}`;

exports[`<RowComponent /> renders correctly 1`] = `ShallowWrapper {}`;

exports[`renders correctly 1`] = `ShallowWrapper {}`;
